DOCS-3658 Adds Use Session Replay In Your Solutions/Support Workflows Guide + Troubleshooting Doc Page (#14445)
DataDog/documentation: content/en/real_user_monitoring/session_replay/_index.md

---
title: Session Replay
kind: documentation
description: Learn about how to capture and visually replay your users' web browsing experience with Session Replay.
further_reading:
- link: 'https://www.datadoghq.com/blog/session-replay-datadog/'
  tag: 'Blog'
  text: 'Use Datadog Session Replay to view real-time user journeys'
- link: 'https://www.datadoghq.com/blog/reduce-customer-friction-funnel-analysis/'
  tag: 'Blog'
  text: 'Use funnel analysis to understand and optimize key user flows'
- link: '/real_user_monitoring/explorer'
  tag: 'Documentation'
  text: 'Visualize your RUM data in the Explorer'
---

## Overview

Session Replay expands your user experience monitoring by allowing you to capture and visually replay the web browsing experience of your users. Combined with RUM performance data, Session Replay is beneficial for error identification, reproduction, and resolution, and provides insights into your web application’s usage patterns and design pitfalls.

The RUM Browser SDK is [open source][1] and leverages the open source [rrweb][2] project.

> ## Session Replay Recorder

Reviewer: ```suggestion
## Session Replay recorder
```

> The Session Replay recorder, part of the RUM Browser SDK, takes a snapshot of the browser's DOM and CSS by tailing and recording events happening on a web page (such as DOM modification, mouse move, clicks, and input events) along with their timestamps. 

Reviewer: ```suggestion
The Session Replay recorder is part of the RUM Browser SDK. The recorder takes a snapshot of the browser's DOM and CSS by tailing and recording events happening on a web page (such as DOM modification, mouse move, clicks, and input events) along with these events' timestamps. 
```

Datadog then rebuilds the web page and re-applies the recorded events at the appropriate time in the replay view. Session Replay follows the same 30 day retention policy as normal RUM sessions. 

The Session Replay recorder supports all browsers supported by the RUM Browser SDK with the exception of IE11. For more information, see the [browser support table][3]. 

> In order to reduce Session Replay's network impact and ensure the Session Replay recorder has minimal overhead on your application's performance, Datadog compresses the data prior to sending it, and reduces the load on a browser’s UI thread by delegating most of the CPU-intensive work (such as compression) to a background service worker. The expected network bandwidth impact is less than 100Kb/min. 

Reviewer: ```suggestion
To reduce Session Replay's network impact and ensure the Session Replay recorder has minimal overhead on your application's performance, Datadog compresses the data prior to sending it. Datadog also reduces the load on a browser’s UI thread by delegating most of the CPU-intensive work (such as compression) to a background service worker. The expected network bandwidth impact is less than 100kB/min. 
```

## Setup

> Session Replay is available through a dedicated build of the RUM Browser SDK. To start collecting data for Session Replay, set up [Datadog RUM Browser Monitoring][4] by creating a RUM application, generating a client token generation, and initializing the RUM Browser SDK.

Reviewer 2: ```suggestion
Session Replay is available in the RUM Browser SDK. To start collecting data for Session Replay, set up [Datadog RUM Browser Monitoring][4] by creating a RUM application, generating a client token generation, and initializing the RUM Browser SDK.
```
Just checked with @DataDog/rum-browser and this sentence isn't accurate anymore. There is no dedicated build, SR is included in the standard bundle/package now.

datadogRum.startSessionReplayRecording();
```

### CDN

Replace the RUM Browser SDK URL `https://www.datadoghq-browser-agent.com/datadog-rum.js` with `https://www.datadoghq-browser-agent.com/datadog-rum-v4.js`. When `DD_RUM.init()` is called, the Session Replay recording does not start until `DD_RUM.startSessionReplayRecording()` is also called.

## Configuration

> The usual [RUM initialization parameters][6] are all supported.

Author: ```suggestion
You can use the RUM Browser SDK's [initialization parameters][6].
```

}
```

To stop the Session Replay recording, call `stopSessionReplayRecording()`.

### Disable Session Replay

> To stop session recordings, remove `startSessionReplayRecording()` and set `replaySampleRate` to `0`. This stops collecting data for the RUM & Session Replay's [Browser Premium plan][7], which includes replays, resources, and long tasks.

Author: ```suggestion
To stop session recordings, remove `startSessionReplayRecording()` and set `replaySampleRate` to `0`. This stops collecting data for RUM & Session Replay's [Browser Premium plan][7], which includes replays, resources, and long tasks.
```

> To stop session recordings, remove `startSessionReplayRecording()` and set `replaySampleRate` to `0`. This stops collecting data for the [Browser Premium plan][7] for RUM & Session Replay, which includes replays, resources, and long tasks.

Author: ```suggestion
To stop session recordings, remove `startSessionReplayRecording()` and set `replaySampleRate` to `0`. This stops collecting data for the RUM & Session Replay's [Browser Premium plan][7], which includes replays, resources, and long tasks.
```

> In order to apply these configurations, upgrade the [Browser RUM SDK][5] to a version >= 3.6.

Author: ```suggestion
In order to apply these configurations, upgrade the [RUM Browser SDK][5] to a version >= 3.6.
```
